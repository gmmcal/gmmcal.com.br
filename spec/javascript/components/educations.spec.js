import { mount, shallowMount } from '@vue/test-utils'
import Educations from '@/components/educations'
import Education from '@/components/education'
import create from '../settings/utils'

describe('Educations', () => {
  let wrapper

  beforeEach(() => {
    window.gon = { educations: create('educations') }

    wrapper = mount(Educations)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['educations', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Child components', () => {
    it('includes Educations', () => {
      const child_component = wrapper.find(Educations)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('renders 2 Education objects', () => {
      wrapper = shallowMount(Educations)
      const educations = wrapper.findAll(Education)
      expect(educations.length).toBe(2)
    })
  })
})
