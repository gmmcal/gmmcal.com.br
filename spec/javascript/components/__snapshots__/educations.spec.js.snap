// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Educations renders correctly 1`] = `
<section id="education" class="education">
  <div class="container">
    <div class="heading">
      <h2>Education and Diplomas</h2>
      <p>What I did on my academic career</p>
    </div>
    <div class="row education-details">
      <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
        <div class="year">Currently</div>
      </div>
      <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
        <div class="arrowpart"></div>
        <div class="ex-con">
          <h4>Some awesome university</h4>
          <h5>Computer science</h5>
          <div class="description">
            <p>Lorem ipsum nulus dolor</p>
          </div>
        </div>
      </div>
    </div>
    <div class="row education-details">
      <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
        <div class="year">Jan / 2016</div>
      </div>
      <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
        <div class="arrowpart"></div>
        <div class="ex-con">
          <h4>Some awesome university</h4>
          <h5>Computer science</h5>
          <div class="description">
            <p>Lorem ipsum nulus dolor</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
